tsdb/engine/tsm1/compact.go

```diff
@@ -581,7 +581,9 @@ func (c *DefaultPlanner) Plan(lastWrite time.Time) ([]CompactionGroup, int64) {
 	end := 0
 	start := 0
 	for i, g := range generations {
-		if g.level() <= 3 {
+		// 4 is used as the step size for our planned compactions, if there are 4 files
+		// in this generation it will get picked up by a level compaction
+		if g.level() <= 3 && len(g.files) > 4 {
```

Comment:
this rule is going to greatly expand what this Plan produces. Consider

`5,5,5,5,5,4,4,4,3,3,3,3,2,2,2,2,3,3,3,3,2,2,2,2,1,1,1,1,2,2,2,2,1,1,1,1,1,1,1`

In master it would plan:
`(5,5,5,5),(5,4,4,4)`

With this pr, it will plan"
`(5,5,5,5),(5,4,4,4),(3,3,3,3),(2,2,2,2),(3,3,3,3),(2,2,2,2),(1,1,1,1),(2,2,2,2)`

It'll only run the first plan if the scheduler picks it: namely `(5,5,5,5)` but because of the additional groups, it'll have significantly more weight.